src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java

```diff
@@ -64,7 +64,7 @@ public void clickReturn() {
 
 	@Then("System returns to patient dashboard")
 	public void returnToDashboard() {
-		matchPatientIds(patientDashboardId);
+		assertPage(dashboardPage.waitForPage());
```

Comment:
So ideally, let's use this example, suppose the system doesn't return to the dashboard page, and your assertion is waiting for dashboard page instead of a functional  method that directly checks on the known element or id , how would you include that assertion

Reply:
When the system doesn't return to the dashboard page the test must fail the run because the run is expected to be on the dashboard page for the next operation to be successful. As well the previous assertion is good enough to assert the return of the system on the dashboard  since it asserts the patientDashboardId instead of a text. In both assertion if the the system fails the to return on the dashboard page, the test must fail!

Reply:
Great, so ideally you are saying the best approach would assertPage instead of text on the page, i do believe this approach also does the same thing, However i havent seen any logic behind this kind of approach, since we might populate assertPage in assertion area, this might look like not clean code, i do believe using text() still would bring the implementation more sufficient, Am not objecting the idea, but am trying to look at the idea that is going to be more appropriate

Reply:
Since we are dealing with loading pages, it would better to rather affirm the page is loaded  successfully using `assertPage() `than using the `assertText()` on the page. If we use texts, then the tests will fail the run in case;
1. the text in the assertion has been changed by UI designers.
2. the locale is changed from English.

Reply:
@kdaud what happens when the page loads without the expected test? Will the test still fail as expected?

Reply:
> @kdaud what happens when the page loads without the expected test? Will the test still fail as expected?

Have stepped back over the question and realistically the test will pass without the expected test. This would mean resorting to asserting texts on the page however, am skeptical  over asserting tests via texts, in case the locale is changed from English all the tests will fail. Unless we assume this scenario not to happen any way.

Reply:
Is asserting text the only way we can reliably locate it?

Reply:
@sherrif10 is there an alternative for asserting texts for loading system page?

Reply:
> Is asserting text the only way we can reliably locate it?

@dkayiwa have not yet known another way but hope will get a better way, for now we can resort to asserting texts.